### PR TITLE
138 Create unit test to package mapping 26-30

### LINKED ISSUE #138: Create unit test to package Mapping 26-30
- MultipartBase64ImageMapper
- NewTagDtoMapper
- SearchNewsDtoMapper
- ShoppingListItemDtoMapper
- ShoppingListItemRequestDtoMapper

### PR DESCRIPTION
**Pull request by Denys8807**
## Summary by CodeRabbit

* **Tests**
  * Added new test coverage for image-to-file conversion, tag-to-DTO mapping, locale-sensitive news DTO mapping, shopping list item DTO mapping, and shopping list item request DTO mapping. These tests verify correct mapping logic, error handling, and locale-specific behavior across multiple components.

### CODE REVIEW

File: service/src/test/java/greencity/mapping/MultipartBase64ImageMapperTest.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+        assertNotNull(result);
+        assertEquals("mainFile", result.getName());
+        assertEquals("tempImage.jpg", result.getOriginalFilename());
+        assertTrue(result.getSize() > 0);
+    }
+
+
+    @Test
+    void convert_ValidImage_ShouldCleanupTempFile() {
```
Comment (natri0) [inline service/src/test/java/greencity/mapping/MultipartBase64ImageMapperTest.java]: it doesn't really test if the mapper cleans the file up, though?

```suggestion
    void convert_ValidImage_ShouldCreateTempFile() {
```

(this is the same suggestion that rabbit did, lol.)

File: service/src/test/java/greencity/mapping/NewTagDtoMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+                .tag(tag)
+                .build();
+        tag.setTagTranslations(Arrays.asList(englishTranslation1, englishTranslation2));
+
+        NewTagDto result = mapper.convert(tag);
+
+        assertNotNull(result);
+        assertEquals(1L, result.getId());
+        assertEquals("First English", result.getName());
+        assertNull(result.getNameUa());
```
Comment (natri0) [inline service/src/test/java/greencity/mapping/NewTagDtoMapperTest.java]: this assert doesn't really need to be here as we already test that no translation = null above.

File: service/src/test/java/greencity/mapping/ShoppingListItemDtoMapperTest.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    @InjectMocks
+    private ShoppingListItemDtoMapper shoppingListItemDtoMapper;
+    private ShoppingListItemTranslation shoppingListItemTranslation;
+    private ShoppingListItem shoppingListItem;
+
+    @BeforeEach
+    void setUp() {
+        shoppingListItem = mock(ShoppingListItem.class);
+        shoppingListItemTranslation = mock(ShoppingListItemTranslation.class);
+    }
```
Comment (natri0) [inline service/src/test/java/greencity/mapping/ShoppingListItemDtoMapperTest.java]: could use `@Mock` annotation


```suggestion
    @Mock
    private ShoppingListItemTranslation shoppingListItemTranslation;
    @Mock
    private ShoppingListItem shoppingListItem;
```

File: service/src/test/java/greencity/mapping/ShoppingListItemDtoMapperTest.java [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+        when(shoppingListItemTranslation.getShoppingListItem()).thenReturn(shoppingListItem);
+        when(shoppingListItem.getId()).thenReturn(expectedId);
+        when(shoppingListItemTranslation.getContent()).thenReturn(expectedText);
+
+        ShoppingListItemDto result = shoppingListItemDtoMapper.convert(shoppingListItemTranslation);
+
+        assertNotNull(result);
+        assertEquals(ShoppingListItemStatus.ACTIVE.toString(), result.getStatus());
+        assertEquals("ACTIVE", result.getStatus());
```
Comment (natri0) [inline service/src/test/java/greencity/mapping/ShoppingListItemDtoMapperTest.java]: duplicated assert?

File: service/src/test/java/greencity/mapping/ShoppingListItemRequestDtoMapperTest.java [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+        Long expectedId = Long.MAX_VALUE;
+        requestDto.setId(expectedId);
+
+        UserShoppingListItem result = mapper.convert(requestDto);
+
+        assertNotNull(result);
+        assertNotNull(result.getShoppingListItem());
+        assertEquals(expectedId, result.getShoppingListItem().getId());
+        assertEquals(ShoppingListItemStatus.ACTIVE, result.getStatus());
+    }
```
Comment (natri0) [inline service/src/test/java/greencity/mapping/ShoppingListItemRequestDtoMapperTest.java]: i'm pretty sure those three tests could be turned into a parameterized one

File: service/src/test/java/greencity/mapping/ShoppingListItemRequestDtoMapperTest.java [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+    void convert_ShouldCreateBuilderBasedObjects() {
+        requestDto.setId(100L);
+
+        UserShoppingListItem result = mapper.convert(requestDto);
+
+        assertNotNull(result);
+        assertNotNull(result.getShoppingListItem());
+        assertNotNull(result.getShoppingListItem().getId());
+        assertNotNull(result.getStatus());
+    }
```
Comment (natri0) [inline service/src/test/java/greencity/mapping/ShoppingListItemRequestDtoMapperTest.java]: this test is redundant

**Review by natri0 — CHANGES_REQUESTED** [review]

**Comment by Denys8807** [comment]
@natri0 Thanks a lot for the thorough review! I’ve made the fixes you pointed out.

**Review by natri0 — APPROVED** [review]

**Review by oleksandrYeriemieiev — APPROVED** [review]

**Review by AndriiSynytsia — APPROVED** [review]
